### PR TITLE
Support OpeningHoursSpecification structured data for creating OpeningHours

### LINKED ISSUE #231: [Feature request] createFromStructuredData method
Hello,

Is there any plan to construct `OpeningHours` object **from** https://schema.org/OpeningHoursSpecification ?
If not, will you accept pull-request ?

I need to store opening-hours data in database then I chose to store as a native `OpeningHoursSpecification` JSON object, but I need to parse it for my business logic.

Maybe you'll have better options.

### PR DESCRIPTION
**Pull request by ambroisemaupate**
- Adds `OpeningHoursSpecificationParser` final class with its unit-test 
- Adds `OpeningHours::createFromStructuredData` method.

Fixes #231

### CODE REVIEW

File: src/OpeningHoursSpecificationParser.php [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+            ! preg_match('/^\d{2}:\d{2}(:\d{2})?$/', $closes)
+        ) {
+            throw new InvalidOpeningHoursSpecification('Invalid opens and closes hours');
+        }
+
+        // strip seconds part if present
+        $opens = preg_replace('/^(\d{2}:\d{2})(:\d{2})?$/', '$1', $opens);
+        $closes = preg_replace('/^(\d{2}:\d{2})(:\d{2})?$/', '$1', $closes);
+
+        // Ignore 00:00-00:00 which means closed all day
```
Comment (kylekatarnls) [inline src/OpeningHoursSpecificationParser.php:169]: This sounds quite unintuitive, where is it specified?

Comment (ambroisemaupate) [reply src/OpeningHoursSpecificationParser.php:169]: Are you talking about `00:00-00:00` or stripping seconds from specification ?

Comment (ambroisemaupate) [reply src/OpeningHoursSpecificationParser.php:169]: 1. OpeningHoursSpecification can support seconds: \
image

2. For `00:00-00:00` it's specified on Google documentation : https://developers.google.com/search/docs/appearance/structured-data/local-business?hl=fr#business-hours

Comment (kylekatarnls) [reply src/OpeningHoursSpecificationParser.php:169]: Ah OK, I was searching in https://schema.org/OpeningHoursSpecification

Now I also understand why you called it Google-flavored in the doc.

**Review by kylekatarnls — APPROVED** [review]

**Pull request merged**